app/protocol/flows/v5/blockrelay/ibd_with_headers_proof.go

```diff
@@ -45,10 +46,26 @@ func (flow *handleIBDFlow) ibdWithHeadersProof(highHash *externalapi.DomainHash,
 	return nil
 }
 
-func (flow *handleIBDFlow) shouldSyncAndShouldDownloadHeadersProof(highBlock *externalapi.DomainBlock,
-	highestSharedBlockFound bool) (shouldDownload, shouldSync bool, err error) {
+func (flow *handleIBDFlow) shouldSyncAndShouldDownloadHeadersProof(
+	highBlock *externalapi.DomainBlock,
+	highestSharedBlockHash *externalapi.DomainHash) (shouldDownload, shouldSync bool, err error) {
 
-	if !highestSharedBlockFound {
+	var highestSharedBlockFound, isPruningPointInSharedBlockChain bool
+	if highestSharedBlockHash != nil {
+		highestSharedBlockFound = true
+		pruningPoint, err := flow.Domain().Consensus().PruningPoint()
+		if err != nil {
+			return false, false, err
+		}
+
+		isPruningPointInSharedBlockChain, err = flow.Domain().Consensus().IsInSelectedParentChainOf(
```

Comment:
If this is false, doesn't it mean we have a finality conflict?

Reply:
This is exactly the same case where with previous algorithm we got shared (chain) block not found